### PR TITLE
Slight cleanup of dead{.h,.cpp}

### PR DESCRIPTION
**Pull request by vlolteanu**
The only outright intrusive change is the use of an std::array for the animation data.

### CODE REVIEW

**Review by AJenbo — APPROVED** [review]
Thanks

File: Source/dead.cpp [hunk 1]
```diff
@@ -17,25 +17,22 @@ int8_t stonendx;
...
-			dead[nd]._deadWidth = Monsters[i].width;
-			dead[nd]._deadtrans = 0;
-			nd++;
+		if (mtypes[Monsters[i].mtype] != 0)
+			continue;
 
-			Monsters[i].mdeadval = nd;
-			mtypes[Monsters[i].mtype] = nd;
-		}
+		dead[nd]._deadData = Monsters[i].Anims[MA_DEATH].Data;
```
Comment (julealgon) [inline Source/dead.cpp:28]: Is this really equivalent to the old 8 position loop?

File: Source/dead.cpp [hunk 2]
```diff
@@ -72,21 +68,21 @@ void InitDead()
 	assert(nd <= MAXDEAD);
 }
 
-void AddDead(int dx, int dy, int8_t dv, direction ddir)
+void AddDead(Point loc, int8_t dv, direction ddir)
```
Comment (julealgon) [inline Source/dead.cpp:71]: Can we rename the parameter to something more descriptive, like `tileLocation` or `tilePosition` or even `tileCoordinate`?

FAKE EDIT: `tilePosition` seems like the best candidate considering our naming for similar values on the `PlayerStruct`.

File: Source/dead.h [hunk 3]
```diff
@@ -5,16 +5,17 @@
...
 
+#include <array>
 #include <cstdint>
 
 #include "engine.h"
 
 namespace devilution {
 
-#define MAXDEAD 31
+static constexpr unsigned MAXDEAD = 31;
```
Comment (julealgon) [inline Source/dead.h:15]: We should not be using ALL_CAPS for these types of constants.

File: Source/dead.h [hunk 4]
```diff
@@ -5,16 +5,17 @@
...
 
 struct DeadStruct {
-	uint8_t *_deadData[8];
+	std::array<uint8_t *, 8> _deadData;
```
Comment (julealgon) [inline Source/dead.h:18]: Oh, so this is what enabled the copy on assignment huh.

**Review by julealgon — CHANGES_REQUESTED** [review]

File: Source/dead.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/dead.cpp:28]: yes, since the type was changed.

File: Source/dead.h (hunk 4 above)
Comment (AJenbo) [reply Source/dead.h:18]: Please resolve the comments when the question isn't relevant anymore.

Comment (julealgon) [reply Source/dead.h:18]: I cannot resolve my comments. Maybe I don't have enough permissions?

Comment (AJenbo) [reply Source/dead.h:18]: ok, will look into that.

File: Source/dead.h (hunk 3 above)
Comment (AJenbo) [reply Source/dead.h:15]:
```suggestion
static constexpr unsigned MaxDead = 31;
```

File: Source/dead.cpp (hunk 2 above)
Comment (AJenbo) [reply Source/dead.cpp:71]:
```suggestion
void AddDead(Point tilePosition, int8_t dv, direction ddir)
```

File: Source/dead.h (hunk 3 above)
Comment (galaxyhaxz) [reply Source/dead.h:15]: one word a.k.a `define` isn't 133t enough in 2021. Instead we need
`static constexpr unsigned uint8_t signed halfsigned` :P

File: Source/dead.cpp (hunk 2 above)
Comment (vlolteanu) [reply Source/dead.cpp]: Done

File: Source/dead.h (hunk 3 above)
Comment (vlolteanu) [reply Source/dead.h]: Done.

@galaxyhaxz Heh, that's true. However,  if you want to keep everything in the devilution namespace, you have to ditch almost all of your macros.

Comment (galaxyhaxz) [reply Source/dead.h]: @vlolteanu Ah, I'm in the process of learning C++ so wasn't sure the exact reason for these changes. Thanks for clarifying :)

**Comment by vlolteanu** [comment]
This is slightly off-topic, but the array used for the animation data shows up in a lot of other places. Maybe it should be typedef-ed, e.g. using AnimationData = std::array<uint8_t, 8>, or something like that.

**Comment by AJenbo** [comment]
Sounds like an idea

**Pull request merged**